services/data-info/src/data_info/util/irods.clj

```diff
@@ -11,8 +11,19 @@
             [data-info.util.config :as cfg])
   (:import [clojure.lang IPersistentMap]
            [java.util UUID]
+           [java.io IOException]
+           [org.irods.jargon.core.exception JargonException]
            [org.apache.tika Tika]))
 
+(defmacro catch-jargon-io-exceptions
+  [& body]
+  `(try+
+     (do ~@body)
+     (catch JargonException e#
+       (if (instance? IOException (.getCause e#))
+         (throw+ {:error_code error/ERR_UNAVAILABLE
+                  :reason (str "iRODS is unavailable: " e#)})
```

Comment:
this "reason" is what will be passed in the JSON response, but the entire stacktrace will be in the logs, in case that's concerning anyone